File: coordinates/src/main/point-2d.ts
export class Point2d {
    public x: number;
    public y: number;

    constructor(x: number = 0, y: number = 0) {
        this.x = x;
        this.y = y;
    }

    public clone(): Point2d {
        let retVal = new Point2d();
        retVal.x = this.x;
        retVal.y = this.y;
        return retVal;
    }

    public calculateDistance(o: Point2d): number {
        return Math.sqrt((this.x - o.x) ** 2 + (this.y - o.y) ** 2)
    }
}

File: vfr-green-app/app/src/main/java/com/blueskycharts/app/coordinates/Point2d.kt
package com.blueskycharts.app.coordinates

import kotlin.math.pow
import kotlin.math.sqrt

open class Point2d(var x: Double = 0.0, var y: Double = 0.0) {
    public fun clone(): Point2d {
        return Point2d(x, y);
    }

    public fun calculateDistance(o: Point2d): Double {
        return sqrt((this.x - o.x).pow(2) + (this.y - o.y).pow(2));
    }

    fun createBoxAround(size: Point2d): Box2d {
        return Box2d(this.x - (size.x / 2.0), this.y - (size.y / 2.0), this.x + (size.x / 2.0), this.y + (size.y / 2.0))
    }
}
